openapi.yaml

```diff
@@ -25,254 +37,677 @@ security:
-     schema:
-       $ref: '#/definitions/ErrorResult'
-securityDefinitions:
-  OAuth2:
-    type: oauth2
-    flow: application
-    tokenUrl: https://api.oregonstate.edu/oauth2/token
-    scopes:
-      full: Full access to the API
+            type: boolean
+            example: true
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+    delete:
+      description: remove the ingredient with ID ingredientID from the pizza with ID pizzaID
+      tags:
+        - pizza
+        - ingredient
+      parameters:
+        - in: path
+          name: ingredientID
+          required: true
+          type: string
+          description: "ID of the ingredient to add"
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: "ID of a pizza to update"
+      responses:
+        200:
+          description: "Returns true is the resource was deleted"
+          schema:
+            type: boolean
+            example: true
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /pizza/{pizzaID}/dough:
+    get:
+      description: "Get the dough used for a pizza"
+      tags:
+        - pizza
+        - dough
+      parameters:
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: ID of a pizza to retrieve
+      responses:
+        200:
+          description: "A single dough recipe"
+          schema:
+            $ref: "#/definitions/DoughResult"
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /pizzas/{pizzaID}/dough/{doughID}:
+    patch:
+      description: set the dough for the pizza with ID pizzaID to the dough with ID doughID
+      tags:
+        - pizza
+        - dough
+      parameters:
+        - in: path
+          name: doughID
+          required: true
+          type: string
+          description: "ID of the dough to use"
+        - in: path
+          name: pizzaID
+          required: true
+          type: string
+          description: "ID of a pizza to update"
+      responses:
+        200:
+          description: "Returns true is the resource was updated"
+          schema:
+            type: boolean
+            example: true
+        404:
+          $ref: "#/responses/404"
+        500:
+          $ref: "#/responses/500"
+  /doughs:
+    get:
+      description: "Get all dough recipes"
+      tags:
+        - dough
+      parameters:
+        - in: query
+          name: name
+          required: false
+          type: string
+          description: Search doughs by name
+        - in: query
+          name: gramsFlour
+          required: false
+          type: integer
+          description: Search doughs by grams flour used
+        - in: query
+          name: gramsWater
+          required: false
+          type: integer
```

Comment:
Same comment as `ovenTemp` in `/pizzas` endpoint. In addition, should this be a floating-point number? Same thing for other parameters.

Reply:
My sense is integer is fine; I've never seen fractions of grams used for these measurements since most kitchen scales aren't precise enough for it to really matter.